src/coreclr/vm/comsynchronizable.cpp

```diff
@@ -729,6 +711,30 @@ FCIMPL1(void, ThreadNative::Finalize, ThreadBaseObject* pThisUNSAFE)
 }
 FCIMPLEND
 
+// Get whether or not this is a background thread.
+extern "C" BOOL QCALLTYPE ThreadNative_GetIsBackground(QCall::ThreadHandle thread)
+{
+    CONTRACTL
+    {
+        QCALL_CHECK;
+        PRECONDITION(thread != NULL);
+    }
+    CONTRACTL_END;
+
+    BOOL res = FALSE;
+
+    BEGIN_QCALL;
+
+    if (ThreadIsDead(thread))
```

Comment:
This check can be on the managed side, and then this can be turned into `QCALL_CHECK_NO_GC_TRANSITION`.

Reply:
Ugh! I forgot I added the `_isDead`... le sigh...